docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +79,36 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an object option.
+
+When `reservedFirst` is defined, React reserved props (`children`, `dangerouslySetInnerHTML`, `key`, `ref`) must be listed before all other props, but still respecting the alphabetical order:
```

Comment:
actually now that i think of it; isn't `dangerouslySetInnerHTML` a special prop only on DOM nodes?

I think you may want separate lists for DOM nodes, and custom components.